libs/api/ebpf_api.cpp

```diff
@@ -1424,19 +1426,22 @@ ebpf_object_get(_In_z_ const char* path) NO_EXCEPT_TRY
     std::copy(path, path + path_length, request->path);
     auto result = invoke_ioctl(request_buffer, reply);
     if (result != ERROR_SUCCESS) {
-        EBPF_RETURN_FD(ebpf_fd_invalid);
+        *fd = (fd_t)ebpf_fd_invalid;
+        EBPF_RETURN_RESULT(win32_error_code_to_ebpf_result(result));
     }
 
     ebpf_assert(reply.header.id == ebpf_operation_id_t::EBPF_OPERATION_GET_PINNED_OBJECT);
 
     ebpf_handle_t handle = reply.handle;
-    fd_t fd = _create_file_descriptor_for_handle(handle);
-    if (fd == ebpf_fd_invalid) {
+    *fd = _create_file_descriptor_for_handle(handle);
```

Comment:
I believe _open_osfhandle sets errno.  Hence you could update the _create_file_descriptor_for_handle() prototype to return the EBPF error rather than simply assuming it's EBPF_NO_MEMORY.

Reply:
For coherence, I preserved the same behavior for all `_create_file_descriptor_for_handle` calls across the rest of the code, all returning EBPF_NO_MEMORY.